vignettes/articles/04-advanced-data-settings.Rmd

```diff
@@ -0,0 +1,101 @@
+---
+title: "04 - Improve stress test by providing further project specific data"
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignettes describes in detail how an analysis can be configured further by
+providing further custom data. The configurations are not necessary but, in case
+required information is available, may help you improve the accuracy of your
+results.
+
+## Prerequisites
+In order to keep this vignette concise, we assume that you have successfully
+complete all preparation steps described in the vignettes ["Set up project
+directories"](https://2degreesinvesting.github.io/r2dii.climate.stress.test/articles/articles/01-setup-project-directories.html)
+and [prepare loans
+inputs](https://2degreesinvesting.github.io/r2dii.climate.stress.test/articles/articles/02-prepare-loans-inputs.html).
+
+
+## Company - term data
+Including company - term data allows you to incorporate the information you have
+on the term values of the companies in your portfolio, instead of using an
+identical term value for all companies.
+
+### Preparing company - term data
+The data that needs to be created is a csv file named `company_terms.csv`. It
+holds the columns `company_name` (character, holding name of analysed companies)
+and term (numeric, indicates which maturity the loans to respective companies is
+assumed to have in years). In column `company_name` include all unique company
+names from `Loans_results_company.rda`, which you generated in [prepare loans
+inputs](https://2degreesinvesting.github.io/r2dii.climate.stress.test/articles/articles/02-prepare-loans-inputs.html).
+If you want to use R to obtain this column you can take the following steps.
+
+```{r, eval = FALSE}
+library(readr)
+library(r2dii.climate.stress.test)
+
+loans_results_company_file_path <- file.path("/path/to/specific/data/inputs", "Loans_results_company.rda")
+validate_file_exists(loans_results_company_file_path)
+data <- readr::read_rds(loans_results_company_file_path)
+unique_companies <- data %>%
+  dplyr::select(company_name) %>%
+  dplyr::distinct()
+readr::write_csv(unique_companies, file.path("/some/path", "company_terms.csv"))
+
+```
+
+You can now open this file in a tool of your choice, and add second column
+named term. In column `term` add the terms per company.   
+
+**NOTE:** 
+
+* Terms need to be whole numbers. 
+* Only include values up to 5. Higher values will be capped to 5 as predictions
```

Comment:
Suggestion:
Only include values up to 5. Higher values will be capped to 5 as for calculations
with longer time frames the reliability of the Merton model deteriorates.

Reply:
done, also changed for conditions in code